kernel/src/hil/hmac.rs

```diff
@@ -0,0 +1,39 @@
+//! Interface for HMAC
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    /// On error `result` will contain device specific error codes
+    fn hash_done(&'a self, result: Result<(), u32>);
+}
+
+pub trait HMAC<'a> {
```

Comment:
I think we could probably make this trait generic over the underlying hash function and thereby avoid having to to set a `SHA256` mode and can return a fixed-size buffer with length determined at compile time with `get_digest()`. I could build something like that as a PR to your fork if you'd like.

Reply:
How could we pass that fixed sized buffer to userspace though?

Reply:
Passing to userspace is an issue I'm not particularly concerned with in the HIL. Please have a look at #1727 to see what I mean by abstracting over a generic type. Still waiting for feedback on that though.